File: CORREDOR/Tipos Constructores/C#/MiProyecto/Telefono.cs
public class Telefono : Electronico, Dispositivo{
    public string Modelo { get; set; }
    //Constructor Sobrecargado
    public Telefono() : base(){
        Modelo = "Desconocido";
    }
    public Telefono(string marca, string modelo) : base(marca){
        Modelo = modelo;
    }
    public void Encender(){
        Console.WriteLine("Encendiendo teléfono...");
    }
    public override void Informacion(){
        Console.WriteLine($"Marca: {Marca}");
        Console.WriteLine($"Modelo: {Modelo}");
    }
}

File: CORREDOR/Tipos Constructores/Go/Telefono.go
package main

import "fmt"

type Telefono struct {
	Electronico
	Modelo string
	Precio float64
}

// Constructor 1
func NewTelefono(modelo string) Telefono {
	return Telefono{
		Electronico: NewElectronico("Genérica"),
		Modelo:     modelo,
		Precio:     0,
	}
}

// Constructor 2 (sobrecargado)
func NewTelefonoFull(marca, modelo string, precio float64) Telefono {
	return Telefono{
		Electronico: NewElectronico(marca),
		Modelo:     modelo,
		Precio:     precio,
	}
}

func (s Telefono) Encender() {
	fmt.Println("Encendiendo el smartphone...")
}

func (s Telefono) Informacion() {
	fmt.Println("Marca:", s.Marca)
	fmt.Println("Modelo:", s.Modelo)
	fmt.Println("Precio:", s.Precio)
}